scripts/docker_cuda_cmake

```diff
@@ -4,16 +4,24 @@ EXTRA_ARGS=$@
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source ${SCRIPT_DIR}/set_kokkos_env.sh
 
-export CUDA_CXX_FLAGS="-g -arch=${GPU_ARCH} -lineinfo \
-    -Xcudafe --diag_suppress=conversion_function_not_usable \
-    -Xcudafe --diag_suppress=cc_clobber_ignored \
-    -Xcudafe --diag_suppress=code_is_unreachable"
+CUDA_NVCC_FLAGS="\"-arch=${GPU_ARCH} --expt-extended-lambda code_is_unreachable \
+-Xcudafe --diag_suppress=conversion_function_not_usable \
+-Xcudafe --diag_suppress=cc_clobber_ignored \
+-Xcudafe --diag_suppress=code_is_unreachable\""
```

Comment:
the `Xcudafe` flags don't suppress the warnings for some unknown reason so there is no point in having them here.